src/Illuminate/Foundation/MaintenanceMode.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Illuminate\Foundation;
+
+use function storage_path;
+
+class MaintenanceMode
+{
+    /**
+     * Determine if the application is currently down for maintenance.
+     *
+     * @return bool
+     */
+    public function isDown(): bool
+    {
+        return file_exists($this->getDownFilePath());
+    }
+
+    /**
+     * Determine if the application is currently up.
+     *
+     * @return bool
+     */
+    public function isUp(): bool
+    {
+        return $this->isDown() === false;
+    }
+
+    /**
+     * Take the application down for maintenance.
+     *
+     * @param  array  $payload
+     * @return void
+     */
+    public function down(array $payload): void
+    {
+        file_put_contents(
+            $this->getDownFilePath(),
+            json_encode($payload, JSON_PRETTY_PRINT)
+        );
+    }
+
+    /**
+     * Take the application out of maintenance.
+     *
+     * @return void
+     */
+    public function up(): void
+    {
+        if ($this->isDown() !== false) {
```

Comment:
`isDown` method always returns a bool value. Why don't you use just `$this->isDown()`?

```php
    public function up(): void
    {
        if ($this->isDown()) {
            unlink($this->getDownFilePath());
        }
    }
```

Reply:
Ah yeah my bad, I said IsUp first but it should indeed be isDown.

Reply:
Good point. Changed it in the code